lib/resourcemerge/batch_test.go

```diff
@@ -0,0 +1,103 @@
+package resourcemerge
+
+import (
+	"testing"
+
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+func TestEnsureJob(t *testing.T) {
+	labelSelector := metav1.LabelSelector{}
+	tests := []struct {
+		name     string
+		existing batchv1.Job
+		required batchv1.Job
+
+		expectedModified bool
+		expected         batchv1.Job
+	}{
+		{
+			name: "different backofflimit count",
+			existing: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(2)}},
+			required: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(3)}},
+
+			expectedModified: true,
+			expected: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(3)}},
+		},
+		{
+			name: "same backofflimit count",
+			existing: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(2)}},
+			required: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(2)}},
+
+			expectedModified: false,
+			expected: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: int32Ptr(2)}},
+		},
+		{
+			name: "required-Selector not nil",
+			existing: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					Selector:       &labelSelector,
+					ManualSelector: boolPtr(false)}},
+			required: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					Selector:       &labelSelector,
+					ManualSelector: boolPtr(true)}},
+
+			expectedModified: false,
+			expected: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					Selector:       &labelSelector,
+					ManualSelector: boolPtr(false)}},
+		},
+	}
+
+	for i, test := range tests {
+		if i == len(tests)-1 {
+			// This test should cause a panic
+			t.Run(test.name, func(t *testing.T) {
+				defer func() {
+					if r := recover(); r == nil {
+						t.Errorf(test.name + " should have panicked!")
+					}
+				}()
+				modified := pointer.BoolPtr(false)
+				EnsureJob(modified, &test.existing, test.required)
+			})
+		} else {
+			t.Run(test.name, func(t *testing.T) {
+				modified := pointer.BoolPtr(false)
+				EnsureJob(modified, &test.existing, test.required)
+				if *modified != test.expectedModified {
+					t.Errorf("mismatch modified got: %v want: %v", *modified, test.expectedModified)
+				}
+
+				if !equality.Semantic.DeepEqual(test.existing, test.expected) {
+					t.Errorf("mismatch Job got: %v want: %v", test.existing, test.expected)
+				}
+			})
+		}
+	}
+}
+
+func int32Ptr(i int32) *int32 {
```

Comment:
You can use `pointer`, which you already import, instead of defining this and your later `boolPtr` yourself.  [Docs][1]. 

[1]: https://godoc.org/k8s.io/utils/pointer#Int32Ptr

Reply:
Fixed. I copied the original code from apps_test.go which I guess is OBE due to updates to pointer package perhaps.

Reply:
6b58e7a uses `pointer` functions.